Libs/BuildingBlocks/Logging/Extension.cs

```diff
@@ -0,0 +1,73 @@
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Serilog;
+using Serilog.Events;
+using Serilog.Exceptions;
+using Serilog.Sinks.SpectreConsole;
+
+namespace BuildingBlocks.Logging;
+
+public static class Extension
+{
+    public static WebApplicationBuilder UseCustomSerilog(this WebApplicationBuilder builder, IWebHostEnvironment env)
+    {
+        builder.Host.UseSerilog((context, _, loggerConfiguration) =>
+        {
+            ApplySerilogDefaultConfigureLogger(context.Configuration, loggerConfiguration, env.WebRootPath);
+        });
+
+        return builder;
+    }
+
+    public static IServiceCollection AddCustomSerilog(this IServiceCollection services, IWebHostEnvironment env)
+    {
+        services.AddSerilog((serviceProvider, loggerConfiguration) =>
+        {
+            var configuration = serviceProvider.GetService<IConfiguration>();
+            ArgumentNullException.ThrowIfNull(configuration);
+            ApplySerilogDefaultConfigureLogger(configuration, loggerConfiguration, env.WebRootPath);
+        });
+
+        return services;
+    }
+
+    private static void ApplySerilogDefaultConfigureLogger(
+        IConfiguration configuration,
+        LoggerConfiguration loggerConfiguration,
+        string rootPath)
+    {
+        var logOptions = configuration.GetSection(nameof(LogOptions)).Get<LogOptions>();
+
+        var logLevel = Enum.TryParse<LogEventLevel>(logOptions!.Level, true, out var level)
+            ? level
+            : LogEventLevel.Information;
+
+        loggerConfiguration
+            .MinimumLevel.Is(logLevel)
+            .WriteTo.SpectreConsole(logOptions.LogTemplate, logLevel)
+            .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
+            .MinimumLevel.Override("Microsoft.EntityFrameworkCore", LogEventLevel.Error)
+            .MinimumLevel.Override("Microsoft.AspNetCore", LogEventLevel.Warning)
+            .MinimumLevel.Override("Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware", LogEventLevel.Fatal)
+            .Enrich.WithExceptionDetails()
+            .Enrich.FromLogContext()
+            .ReadFrom.Configuration(configuration);
+
+        if (logOptions.File.Enabled)
+        {
+            Directory.CreateDirectory(Path.Combine(rootPath, "logs"));
+
+            var path = string.IsNullOrWhiteSpace(logOptions.File.Path) ? "logs/.txt" : logOptions.File.Path;
```

Comment:
```suggestion
            var path = string.IsNullOrWhiteSpace(logOptions.File.Path) ? "logs/logs.txt" : logOptions.File.Path;
```

А вообще, может дефолтом сделать в LogOptions?

Reply:
не совсем понял, в LogOptions дефолтный путь?

Reply:
да

Reply:
тут хз, Options - паттерн в .net, https://learn.microsoft.com/en-us/dotnet/core/extensions/options, для конфигурации. Тут можно вынести отдельно например дефолтные пути, и если в оптионс не передано, то ставить его

Reply:
Даже, наверное, не вынести, а просто константой в файле сделать и лучше будет

Reply:
я вот что-то такое имею в виду
`public string Path { get; set; } = "some/path.txt"`

Reply:
Да, я понял, сначала как то не оч хотел это менять, хотелось явно указывать, но сейчас подумал, норм, что то вроде наследования конфигурации будет